jumpscale/packages/vdc_dashboard/bottle/vdc_helpers.py

```diff
@@ -1,10 +1,32 @@
 from bottle import HTTPResponse, abort
-import math
 from jumpscale.loader import j
 
 from jumpscale.sals.vdc.size import VDC_SIZE
 from jumpscale.sals.vdc.models import KubernetesRole
 
+CURRENTVDC = None
+
+
+def require_vdc_info(fun):
+    def inner(*args, **kwargs):
+        global CURRENTVDC
+        if not CURRENTVDC:
+            CURRENTVDC = get_vdc()
+        return fun(*args, **kwargs)
+
+    return inner
+
+
+def get_vdc(load_info=False):
+    global CURRENTVDC
+    if not j.sals.vdc.list_all():
+        abort(500, "Couldn't find any vdcs on this machine, Please make sure to have it configured properly")
+    vdc_full_name = list(j.sals.vdc.list_all())[0]
+    if not CURRENTVDC or load_info:
+        CURRENTVDC = j.sals.vdc.find(vdc_full_name, load_info=load_info)
+        CURRENTVDC.save()
```

Comment:
why do we do `save()` here?

Reply:
I think by saving the object we can avoid loading info in any future calls or no need to save?